torch_xla/csrc/runtime/pjrt_computation_client.cc

```diff
@@ -185,6 +185,13 @@ ComputationClient::DataPtr PjRtComputationClient::CreateDataPlaceholder(
   return std::make_shared<PjRtData>(std::move(device), std::move(shape));
 }
 
+ComputationClient::DataPtr PjRtComputationClient::CreateData(
+    std::string device, xla::Shape shape,
+    std::shared_ptr<xla::PjRtBuffer> pjrt_buffer) {
+  return std::make_shared<PjRtData>(std::move(device), std::move(shape),
+                                    pjrt_buffer);
+}
```

Comment:
I think you just need a helper, not the class method.

Reply:
The `struct PjRtData` is private to PjRtComputationClient so I can't make it a helper.

Reply:
Does the input `pjrt_buffer` depend on the current instance of `PjRtClient`? ie do you need to check here that this buffer belongs to this client? If you don't depend on any members of `PjRtComputationClient`, can you make this a static method to make that clearer?

Also, this should be private and probably not part of `ComputationClient`. The public API of `PjRtComputationClient` shouldn't operate on PJRT primitives. It should be a complete wrapper to leave the door open to other runtime interfaces (namely IFRT).

Reply:
Right, the `pjrt_buffer` doesn't depend on the current instance of `PjRtClient`. I have made it a static method.

I also made it not a part of `ComputationClient`. Not sure how I can make it private and call it in the dl_convertor.cpp at the same time.